pywinauto/application.py

```diff
@@ -356,10 +356,14 @@ def __getattribute__(self, attr_name):
         else:
             # if we have been asked for an attribute of the dialog
             # then resolve the window and return the attribute
-            if len(self.criteria) == 1 and hasattr(DialogWrapper, attr_name):
+            desktop_wrapper = self.backend.generic_wrapper_class(self.backend.element_info_class())
+            need_to_resolve = (len(self.criteria) == 1 and hasattr(desktop_wrapper, attr_name))
+            if self.backend.name == 'win32':

```

Comment:
Good point. I will try to make even better fix that will fully move this import from `application.py`.